### PR TITLE
Semian gem

### PR DESCRIPTION
**Pull request by supantha**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+source "https://rubygems.org"
+ruby "~> 2.4.1"
+
+git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
```
Comment (azitabh) [inline Gemfile]: why is this required?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,89 @@
...
+
+Add this line to your application's Gemfile:
+
+```ruby
+gem 'configure_semian'
+```
+
+And then execute:
+
+  $ bundle
```
Comment (azitabh) [inline README.md]: isn't it bun bundle install ?

File: configure_semian.gemspec [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  spec.description   = "Semian Connector"
+  spec.homepage      = "TODO: Put your gem's website or public repo URL here."
+  spec.license       = "MIT"
+  spec.add_dependency "rails", ">=4.0.2"
+  spec.add_dependency "semian"
+
+
+  # Prevent pushing this gem to RubyGems.org. To allow pushes either set the 'allowed_push_host'
+  # to allow pushing to a single host or delete this section to allow pushing to any host.
+  if spec.respond_to?(:metadata)
```
Comment (azitabh) [inline configure_semian.gemspec]: since this is a open source project, this will actually be published to rugygems only

File: lib/configure_semian/engine.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module ConfigureSemian
+  class Engine < ::Rails::Engine
+  	# config.autoload_paths += %W(#{config.root}/lib/configure_semian/semian_configuration.rb)
+    # config.autoload_paths += %W(#{config.root}/lib/configure_semian/net_http.rb)
```
Comment (azitabh) [inline lib/configure_semian/engine.rb]: no reason to keep the commented lines

File: lib/configure_semian/net_http.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+require 'net/http'
+require 'byebug'
```
Comment (azitabh) [inline lib/configure_semian/net_http.rb]: no buybug please

File: lib/configure_semian/net_http.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    alias_method :old_request, :request
+    alias_method :old_post, :post
+
+    def request(req, body = nil, &block)
+      timeout = get_request_timeout_value(self.address, req.path)
+      self.read_timeout = timeout
+      old_request(req, body, &block)
+    end
+
+    def get_request_timeout_value(host, path)
```
Comment (azitabh) [inline lib/configure_semian/net_http.rb]: Add test cases against this method... values we get against various ways we can keep the configurations

File: Gemfile (hunk 1 above)
Comment (supantha) [reply Gemfile]: It is auto generated from bundler 2.x. If the path of any gem is given as the git repo, bundler by default tries to get it via http, this line ensures that https is used.
